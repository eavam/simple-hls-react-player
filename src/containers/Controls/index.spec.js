import React from 'react'
import renderer from 'react-test-renderer'
import { Controls } from './index.js'


test('Controls', () => {
  const props = {
    actions: {},
    volume: '30',
    muted: true,
    currentTime: '04:34',
    isFullscreen: false,
    title: 'test title',
    handleChangeVolume: () => {},
    isPause: false,
    videoEl: document.createElement('video')
  }

  const component = renderer.create(
    <Controls {...props} />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()

})

