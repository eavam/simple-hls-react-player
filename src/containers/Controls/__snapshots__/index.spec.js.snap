// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls 1`] = `
<div
  className="controls-wrapper"
>
  <div
    className="top-controls-wrapper"
  >
    <div
      className="title"
    >
      test title
    </div>
  </div>
  <div
    className="bottom-controls-wrapper"
  >
    <div
      className="left-column"
    >
      <div
        className="player-svg-icon icon-margin-right"
        onClick={undefined}
      >
        <svg
          className={undefined}
          viewBox="0 0 20.03 25"
        >
          <path
            d="M7 21.52a3.48 3.48 0 0 1-7 0v-18a3.48 3.48 0 1 1 7 0zm0 0M20 21.52a3.48 3.48 0 0 1-7 0v-18a3.48 3.48 0 0 1 7 0zm0 0"
            fill="#fff"
          />
        </svg>
      </div>
      <div
        className="sound-control"
      >
        <div
          className="player-svg-icon"
          onClick={undefined}
        >
          <svg
            className={undefined}
            viewBox="0 0 27 27"
          >
            <path
              d="M20.45 3.13a1.64 1.64 0 0 0-2.66-1.29L12.52 6a3.19 3.19 0 0 0-.9-.13H9.37l11.08 11.04zM3 9v9.36a3.21 3.21 0 0 0 3.21 3.21h5.41a3.22 3.22 0 0 0 .78-.1l5.39 4.23a1.64 1.64 0 0 0 2.66-1.3V24L3.62 7.16A3.18 3.18 0 0 0 3 9zm23.71 16.29l-25-25A1 1 0 0 0 .29 1.71l25 25a1 1 0 0 0 1.41-1.41z"
              fill="#fff"
            />
          </svg>
        </div>
        <div
          className="volume-control"
          onMouseDown={[Function]}
          style={
            Object {
              "boxShadow": "inset 0px 0px 0 #fff",
            }
          }
        />
      </div>
    </div>
    <div
      className="right-column"
    >
      <div
        className="current-time"
      >
        04:34
      </div>
      <div
        className="player-svg-icon icon-margin-left"
        onClick={undefined}
      >
        <svg
          className={undefined}
          viewBox="0 0 25 25"
        >
          <path
            d="M4.54 20.5h16v-16h-16zM6.83 6.79h11.43v11.42H6.83zm0 0"
            fill="#fff"
          />
          <path
            d="M2.28 18.18H0V25h6.82v-2.28H2.28zm0 0M2.28 2.28h4.54V0H0v6.82h2.28zm0 0M22.72 22.72h-4.54V25H25v-6.82h-2.28zm0 0M18.18 0v2.28h4.54v4.54H25V0zm0 0"
            fill="#fff"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
